tests/mtouch/MTouch.cs

```diff
@@ -2251,6 +2251,28 @@ public void MT2015 ()
 		}
 
 		[Test]
+		public void MT2017 ()
+		{
+			using (var mtouch = new MTouchTool ()) {
+				// Create a library, copy it to a different directory, and then
+				// pass both as -r:.. to mtouch. Due to assembly resolution being cached,
+				// this will *not* show the MT2017 error (in fact I don't know if it's possible
+				// to run into MT2017 at all).
```

Comment:
I guess this is what happens when I have a big local change, try to reduce it to smaller PRs, and then forget why I did some of the things I did :smile:

The most important change here (I remember now :smile:) is that we end up with a dictionary of Assembly instances keyed by the assembly name.

I'll rework this.